### ISSUE TITLE
Rewrite/explode AssetAPI::ParseAssetRef.

### ISSUE THREAD

**Issue by juj** [realXtend/tundra#408]
The AssetAPI ParseAssetRef has become an extremely used utility throughout the whole Tundra codebase. It is called thousands of times for larger scenes. The function eagerly decomposes an asset ref to all its subparts, and uses several regexes to perform the query. In some scenes, this is starting to show up in profiles. (after #379, #332, #334 and #335)

Rewrite the function and split it into smaller utility functions which do less work to optimize the CPU-side work performed here.
